hedera-node/hedera-app/src/main/java/com/hedera/node/app/blocks/impl/streaming/BlockState.java

```diff
@@ -2,30 +2,43 @@
 package com.hedera.node.app.blocks.impl.streaming;
 
 import com.hedera.hapi.block.stream.BlockItem;
-import edu.umd.cs.findbugs.annotations.NonNull;
+import edu.umd.cs.findbugs.annotations.Nullable;
+import java.time.Instant;
 import java.util.ArrayList;
+import java.util.Iterator;
 import java.util.List;
+import java.util.concurrent.atomic.AtomicBoolean;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.hiero.block.api.BlockItemSet;
 import org.hiero.block.api.PublishStreamRequest;
 
 /**
- * Track the current block state
+ * Represents the state of a block being streamed to block nodes. This class maintains the block items,
+ * completion status, and request generation for a specific block number.
+ * The block state goes through the following lifecycle:
+ * <ul>
+ *     <li>Created when a new block is opened</li>
+ *     <li>Block items are added sequentially</li>
+ *     <li>Requests are generated from accumulated items capped by a configurable batch size</li>
+ *     <li>Block is marked as complete after all items including BlockProof are added</li>
+ * </ul>
  */
 public class BlockState {
+    private static final Logger logger = LogManager.getLogger(BlockState.class);
     private final long blockNumber;
-    private final List<BlockItem> items;
-    private final List<PublishStreamRequest> requests;
-    private boolean isComplete;
+    private final List<BlockItem> items = new ArrayList<>();
+    private final List<PublishStreamRequest> requests = new ArrayList<>();
+    private final AtomicBoolean requestsCreated = new AtomicBoolean(false);
```

Comment:
Since all the other collections in this class aren't thread-safe, it's imo surprising to see an `AtomicBoolean` here. 

Would be good to call out the concurrency guarantees that users of this class have to enforce.